File: src/jump_calc.py
#!/usr/bin/env python
# coding:utf-8

INPUT_NAME = 'INPUT.TXT'
OUTPUT_NAME = 'OUTPUT.TXT'


def read_data(path):
    with open(path) as _:
        one_jump, length = list(map(int, _.readline().strip().split()))
    return one_jump, length


def write_result(path, result):
    with open(path, 'w') as _:
        _.write(str(result))


def graph_vars(one_jump, length, steps=0, counter=0):
    if length > steps:
        for next_step_len in range(1, one_jump + 1):
            counter = graph_vars(one_jump, length, steps=steps + next_step_len, counter=counter)
    elif length == steps:
        counter += 1
    return counter


def main():
    one_jump, length = read_data(INPUT_NAME)
    combinations = graph_vars(one_jump, length)
    write_result(OUTPUT_NAME, combinations)


if __name__ == '__main__':
    main()

File: src/jump_calc.c
#include <stdio.h>
#include <time.h>

int graph_vars(int *one_jump, int *length, int steps) {
    static int counter=0;
    if (*length > steps) {
        for (int next_step_len = 1;  next_step_len <= *one_jump; next_step_len++) {
            graph_vars(one_jump, length, steps + next_step_len);
        }
    } else if (*length == steps) {
        counter += 1;
    }
    return counter;
}

int main() {
    int max_steps, all_steps, variants;
    clock_t t;
    char *input_name = "INPUT.TXT";
    char *output_name = "OUTPUT.TXT";
    FILE *input_data, *out_data;

    input_data = fopen(input_name, "r");
    if (input_data == NULL)
    {
        printf("No such file %s", input_name);
        return 1;
    } else {
        fscanf(input_data, "%d %d", &max_steps, &all_steps);
        printf("max steps per jump: %d;\nall steps: %d;\n", max_steps, all_steps);
    }
    fclose(input_data);
    t = clock();
    variants = graph_vars(&max_steps, &all_steps, 0);
    t = clock();
    double time_taken = ((double)t)/CLOCKS_PER_SEC; // in seconds
    printf("variants: %d; \ntook %f seconds\n", variants, time_taken);
    out_data = fopen(output_name,"w");
    fprintf(out_data, "%d", variants);
    fclose(out_data);
    return 0;
}